### PR TITLE
Add xUnit tests for ContactsController

### PR DESCRIPTION
**Pull request by alexashley**
- changed the project structure (2 projects under one "solution")
- added tests for ContactsController using xUnit

I tried out xUnit because there's already a project template for it  (`dotnet new xunit`) and I'm lazy. 
From my first impression, xUnit seems pretty similar to NUnit/MSTest, with the slight exception that there are no methods that run before/after individual tests. You're forced to use the constructor and `Dispose`, but those are on the class-level. 

Also the `Assert` library is a little nicer (`Assert.True` vs `Assert.IsTrue`)

### CODE REVIEW

File: dotnetcorehack.Test/Controllers/ContactsControllerTest.cs [hunk 1]
```diff
@@ -0,0 +1,155 @@
...
+            }
+        }
+
+        public class PostTest : ContactsControllerTest
+        {
+            [Fact]
+            public void ShouldCallRepoWithContact()
+            {
+                var contact = new Contact();
+                _contactRepo.Setup(repo => repo.createContact(contact)).Callback(() => contact.id = _expectedId);
```
Comment (alexashley) [inline dotnetcorehack.Test/Controllers/ContactsControllerTest.cs]: I'd like a second opinion on using `Moq.Callback` in this fashion. It seems to work, but I'm not sure I like mutating the `Contact` argument.

**Comment by dlogankeenan** [comment]
@alexashley, thanks for the PR contribution!!  The PR looks great. I'd like to wait until after work today so I can pull it down and try things out.

File: README.md [hunk 2]
```diff
@@ -10,4 +10,5 @@ A very basic API to get back into things with C#
 ## Run the app 
 1. `dotnet restore` - this will download all the dependnecies
 2. `dotnet run` - this will start the app
-3. JSON request using Postman can be made to `http://localhost:5000/contacts`
+3. JSON request using Postman can be made to `http://localhost:5000/api/contacts`
```
Comment (dlogankeenan) [inline README.md]: I think this shouldn't contain /api/ according to this line of [code](https://github.com/dlogankeenan/dotnetcorehack/blob/master/Controllers/ContactsController.cs#L16).

**Comment by zpratt** [comment]
Thanks for adding this @alexashley.

I did some digging to see if there are any .net libs that work with .net core which have a similar API to mocha: https://github.com/nspec/NSpec

I think it's worth considering. It would mitigate the context switching of test framework APIs.

**Comment by alexashley** [comment]
@zpratt Looks interesting. I wonder if there's any relation to [MSpec](https://github.com/machine/machine.specifications)

 I'll update this PR with some NSpec tests so we can do a side-by-side comparison, but just looking at the combination of NSpec and FluentAssertions has me leaning that way.

**Comment by dlogankeenan** [comment]
@alexashley I'm going to go ahead and merge. Feel free to add another PR

**Pull request merged**